### PR TITLE
Icon: code cleanup, iconName to be deprecated in favor of name prop

### PR DESCRIPTION
**Pull request by dzearing**
This ultimately cleans up Icon code to use the new styling pattern and to remove memoizeFunction usage which causes problems (such as in SSR rendering.) I would like us to remove all usage of memoize, as it fights against SSR but also has unhandled edge cases (such as object and array content comparisons) which won't scale well if we address them.

### CODE REVIEW

File: packages/office-ui-fabric-react/src/components/Icon/Icon.base.tsx [hunk 1]
```diff
@@ -1,103 +1,91 @@
...
 
-export class Icon extends BaseComponent<IIconProps, IIconState> {
+const getClassNames = classNamesFunction<IIconStyleProps, IIconStyles>();
+
+export class IconBase extends BaseComponent<IIconProps, IIconState> {
   constructor(props: IIconProps) {
     super(props);
     this.state = {
       imageLoadError: false,
     };
```
Comment (joschect) [inline packages/office-ui-fabric-react/src/components/Icon/Icon.base.tsx]: Do you still need to add warnDeprecations to alert that iconName will be removed?

Comment (dzearing) [reply packages/office-ui-fabric-react/src/components/Icon/Icon.base.tsx]: I kind of want to do that in 6.0. Icons are used everywhere and warning a kajillion times is going to generate noise. My proposal is that we use `process.env==='production'` checks around warn apis to delete the code in prod builds, but that could break the odsp AMD scenarios...

File: packages/office-ui-fabric-react/src/components/Breadcrumb/__snapshots__/Breadcrumb.test.tsx.snap [hunk 2]
```diff
@@ -202,7 +212,9 @@ exports[`Breadcrumb basic rendering renders breadcumb correctly 1 1`] = `
                   }
               data-icon-name="ChevronRight"
               role="presentation"
-            />
+            >
+              
```
Comment (micahgodbolt) [inline packages/office-ui-fabric-react/src/components/Breadcrumb/__snapshots__/Breadcrumb.test.tsx.snap]: ?

Comment (micahgodbolt) [reply packages/office-ui-fabric-react/src/components/Breadcrumb/__snapshots__/Breadcrumb.test.tsx.snap]: I see ` ` scattered through this PR. Is this something intentional?

Comment (micahgodbolt) [reply packages/office-ui-fabric-react/src/components/Breadcrumb/__snapshots__/Breadcrumb.test.tsx.snap]: ah, this is the snap...nm

**Pull request merged**